### PR TITLE
allergies: updated exercise generator to new test generator

### PR DESCRIPTION
**Pull request by eklatzer**
Hey, here I also adjusted the tests a bit as the inputs seem to have changed

One of #2302

### CODE REVIEW

File: exercises/practice/allergies/allergies_test.go [hunk 1]
```diff
@@ -7,11 +7,13 @@ import (
...
-		if actual := Allergies(test.score); !sameSliceElements(actual, test.expected) {
-			t.Fatalf("FAIL: Allergies(%d): expected %#v, actual %#v", test.score, test.expected, actual)
-		} else {
-			t.Logf("PASS: Allergic to %#v", test.expected)
-		}
+		t.Run(test.description, func(t *testing.T) {
+			if actual := Allergies(test.score); !sameSliceElements(actual, test.expected) {
+				t.Fatalf("FAIL: Allergies(%d): expected %#v, actual %#v", test.score, test.expected, actual)
+			} else {
+				t.Logf("PASS: Allergic to %#v", test.expected)
```
Comment (andrerfcsantos) [inline exercises/practice/allergies/allergies_test.go:14]: With subtests, a string logging a `PASS` or `FAIL` is redundant and can be removed as running `go test -v` will make Go print explicitly the result for every test.

File: exercises/practice/allergies/allergies_test.go [hunk 2]
```diff
@@ -7,11 +7,13 @@ import (
...
 func TestAllergies(t *testing.T) {
 	for _, test := range listTests {
-		if actual := Allergies(test.score); !sameSliceElements(actual, test.expected) {
-			t.Fatalf("FAIL: Allergies(%d): expected %#v, actual %#v", test.score, test.expected, actual)
-		} else {
-			t.Logf("PASS: Allergic to %#v", test.expected)
-		}
+		t.Run(test.description, func(t *testing.T) {
+			if actual := Allergies(test.score); !sameSliceElements(actual, test.expected) {
+				t.Fatalf("FAIL: Allergies(%d): expected %#v, actual %#v", test.score, test.expected, actual)
```
Comment (andrerfcsantos) [inline exercises/practice/allergies/allergies_test.go:12]:
```suggestion
				t.Fatalf("Allergies(%d)=%#v, expected %#v", test.score, actual, test.expected)
```

Removed the `FAIL` string and made the print results be more in line with what we are doing in other exercises.

**Review by andrerfcsantos — CHANGES_REQUESTED** [review]
Made 2 comments that might apply to other exercises.

File: exercises/practice/allergies/allergies_test.go (hunk 1 above)
Comment (eklatzer) [reply exercises/practice/allergies/allergies_test.go]: Should be removed nearly everywhere, might have forgotten/not seen sometimes

**Review by andrerfcsantos — APPROVED** [review]

File: exercises/practice/allergies/allergies_test.go (hunk 1 above)
Comment (andrerfcsantos) [reply exercises/practice/allergies/allergies_test.go]: I'll make sure to double check that, no worries :)

**Pull request merged**